tests/components/homewizard/test_button.py

```diff
@@ -1,174 +1,86 @@
-
-    state = hass.states.get("button.product_name_aabbccddeeff_identify")
-    assert state
-    assert (
-        state.attributes.get(ATTR_FRIENDLY_NAME)
-        == "Product Name (aabbccddeeff) Identify"
-    )
-
-    entity_registry = er.async_get(hass)
-    entry = entity_registry.async_get("button.product_name_aabbccddeeff_identify")
-    assert entry
-    assert entry.unique_id == "aabbccddeeff_identify"
+    assert (entity_entry := entity_registry.async_get(state.entity_id))
+    assert snapshot == entity_entry
 
+    assert entity_entry.device_id
+    assert (device_entry := device_registry.async_get(entity_entry.device_id))
+    assert snapshot == device_entry
 
-async def test_identify_press(
-    hass: HomeAssistant, mock_config_entry_data, mock_config_entry
-) -> None:
-    """Test button press is handled correctly."""
-
-    api = get_mock_device(product_type="HWE-SKT", firmware_version="3.02")
-
-    with patch(
-        "homeassistant.components.homewizard.coordinator.HomeWizardEnergy",
-        return_value=api,
-    ):
-        entry = mock_config_entry
-        entry.data = mock_config_entry_data
-        entry.add_to_hass(hass)
-
-        await hass.config_entries.async_setup(entry.entry_id)
-        await hass.async_block_till_done()
-
-    assert (
-        hass.states.get("button.product_name_aabbccddeeff_identify").state
-        == STATE_UNKNOWN
-    )
-
-    assert api.identify.call_count == 0
+    assert len(mock_homewizardenergy.identify.mock_calls) == 0
     await hass.services.async_call(
         button.DOMAIN,
         button.SERVICE_PRESS,
-        {"entity_id": "button.product_name_aabbccddeeff_identify"},
+        {ATTR_ENTITY_ID: state.entity_id},
         blocking=True,
     )
-    assert api.identify.call_count == 1
+    assert len(mock_homewizardenergy.identify.mock_calls) == 1
 
-
-async def test_identify_press_catches_requesterror(
-    hass: HomeAssistant, mock_config_entry_data, mock_config_entry
-) -> None:
-    """Test button press is handled RequestError correctly."""
-
-    api = get_mock_device(product_type="HWE-SKT", firmware_version="3.02")
-
-    with patch(
-        "homeassistant.components.homewizard.coordinator.HomeWizardEnergy",
-        return_value=api,
-    ):
-        entry = mock_config_entry
-        entry.data = mock_config_entry_data
-        entry.add_to_hass(hass)
-
-        await hass.config_entries.async_setup(entry.entry_id)
-        await hass.async_block_till_done()
-
-    assert (
-        hass.states.get("button.product_name_aabbccddeeff_identify").state
-        == STATE_UNKNOWN
-    )
+    assert (state := hass.states.get(state.entity_id))
+    assert state.state == "2021-01-01T12:00:00+00:00"
 
     # Raise RequestError when identify is called
-    api.identify.side_effect = RequestError()
-
-    assert api.identify.call_count == 0
+    mock_homewizardenergy.identify.side_effect = RequestError()
 
     with pytest.raises(HomeAssistantError):
```

Comment:
We should match the error message as HomeAssistantError is raised quite often

Reply:
It has no error message...